Sources/MongoSwift/MongoError.swift

```diff
@@ -278,6 +278,21 @@ private func getBulkWriteErrorFromReply(
     )
 }
 
+/// Internal function used by write methods that are implemented via the bulk API. Converts a
+/// ServerError.bulkWriteError that resulted from a single write into a ServerError.writeError.
+internal func convertBulkWriteError(_ error: Error) -> Error {
```

Comment:
if you have a better name idea for this please let me know 😛

Reply:
I think the name is fine as-is, but we could consider a different approach altogether to reduce some of the boilerplate of catching and rethrowing.

e.g.
```swift
return try convertBulkWriteErrors {
    let model = InsertOneModel(value)
    let result = try self.bulkWrite([model], options: options?.asBulkWriteOptions())
    return try InsertOneResult(from: result)
}
```
The `convertBulkWriteErrors` function would execute the body and convert any bulk write errors it encounters, otherwise throwing everything else and passing through the return value